### PR TITLE
Add sizes property

### LINKED ISSUE #1634: Extract data extents for a given dimension in python
A bit annoying to have to do this, but not having a method/property to provide it.
```python
va = sc.Variable(['x', 'y'], values=np.arange(100).reshape(20, 5))
dict(zip(va.dims, va.shape))['x']
```
@nvaytet

### PR DESCRIPTION
**Pull request by OwenArnold**
Fixes #1634

### CODE REVIEW

**Review by nvaytet — COMMENTED** [review]
You should also add this to `Dataset`

File: python/bind_data_access.h [hunk 1]
```diff
@@ -358,6 +358,35 @@ using as_ElementArrayView = as_ElementArrayViewImpl<
...
+      for (const auto &dim : dims_.labels()) {
+        out[dim.name()] = dims_[dim];
+      }
+      return out;
+    };
+  }
+  c.def_property_readonly("sizes", generator,
+                          "Dictionary of dimension labels (dims) to sizes.",
+                          py::return_value_policy::move);
+}
```
Comment (SimonHeybrock) [inline python/bind_data_access.h:388]: Seems quite complicated. Would it be simpler to either:
a. Do this on the Python size, by just setting returning `dict(zip(dims, shape))` or
b. Return `Dimensions`, and add bindings for `Dimensions::operator[]`?

Comment (OwenArnold) [reply python/bind_data_access.h:388]: I agree, it's not a nice amount of code to have for something so trivial. Unfortunately `Dataset` and `DatasetView` return a map of dims to indices (such that dimension order is not implied), whereas that is not the case the `DataArray` and `Variable`. I think doing this on the python side would be the best approach. The downside is splitting our property implementations between c++ bindings and monkey-patching python code.

Comment (SimonHeybrock) [reply python/bind_data_access.h:388]: We have split this anyway already, e.g., `_repr_html_` and `bins` are added on the Python side.

I don't see the issue with `Dataset`, it returns an `unordered_map`, so `pybind11` should give you `operator[]` automatically, without need to bind it expicitly.

**Review by SimonHeybrock — APPROVED** [review]

**Pull request merged**